### PR TITLE
uniq documentation 

### LINKED ISSUE #411: Add uniq documentation

### PR DESCRIPTION
**Pull request by umdalema**
# Advanced HW 5 submission

Closes #411 

Name: Dale Mallette
Uniqname: dalema

### CODE REVIEW

File: _commands/basics/uniq.md [hunk 1]
```diff
@@ -2,13 +2,13 @@
 ---
 
 uniq
---
-TODO: Add documentation for this command by submitting a pull request.
-<!-- one line explanation would go here -->
+------
+
+`uniq` allows a user to report or filter repeated lines that are adjacent in a file. If no input or output files are given, uniqu will read and write from standard input or output respectfully.
```
Comment (cameron-gagnon) [inline _commands/basics/uniq.md]: "uniqu" -> "uniq" and you can also surround it with backticks to be `uniq`.

File: _commands/basics/uniq.md [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
-TODO: Add documentation for this command by submitting a pull request.
-<!-- one line explanation would go here -->
+------
+
+`uniq` allows a user to report or filter repeated lines that are adjacent in a file. If no input or output files are given, uniqu will read and write from standard input or output respectfully. 
 
 <!-- minimal example -->
 ~~~ bash
-*short* example of command usage and output
+uniq [options] in.txt out.txt
```
Comment (cameron-gagnon) [inline _commands/basics/uniq.md]: It's standard to prefix commands with a '$ ' (and a space after the dollar sign) so it's known that the command can be run on the command line.

File: _commands/basics/uniq.md [hunk 3]
```diff
@@ -2,13 +2,13 @@
...
 ~~~
```
Comment (cameron-gagnon) [inline _commands/basics/uniq.md]: We're looking for these pull requests to add a good amount of documentation to the commands. Take a look at the other pull requests here, or, go to [c4cs.github.io/reference](https://c4cs.github.io/reference) and click on any of those commands to see how filled out the pages are.

**Comment by umdalema** [comment]
I made several changes and added more to it. Let me know if more needs to be added still.

File: _commands/basics/uniq.md [hunk 4]
```diff
@@ -2,23 +2,39 @@
...
+~~~ bash
+$ sort in.txt | uniq -c 
+  2 line1
+  3 line2
+  1 line3
+~~~
 
 ##### Break it down
+You can also use uniq to list only lines that are duplicates (-d flag) or that are unique (-u flag) 
+##### Break it down
```
Comment (cameron-gagnon) [inline _commands/basics/uniq.md]: Need a newline before the "`##### ...`" for it to render properly.

File: _commands/basics/uniq.md [hunk 5]
```diff
@@ -2,23 +2,39 @@
...
 #### Example command
+~~~ bash
+$ sort in.txt | uniq -c 
+  2 line1
+  3 line2
+  1 line3
+~~~
 
 ##### Break it down
+You can also use uniq to list only lines that are duplicates (-d flag) or that are unique (-u flag)
```
Comment (cameron-gagnon) [inline _commands/basics/uniq.md]: "-d" needs backticks around it so it renders as code. ex: "-d" -> "`-d`"

Same thing for "`-u`"

File: _commands/basics/uniq.md [hunk 6]
```diff
@@ -2,23 +2,39 @@
...
+$ sort in.txt | uniq -c 
+  2 line1
+  3 line2
+  1 line3
 ~~~
 
 <!--more-->
 
 ### Useful Options / Examples
+Often used with sort, uniq will count the number of times a unique line appears in a file when given the -c flag
```
Comment (cameron-gagnon) [inline _commands/basics/uniq.md]: "-c" needs backticks around it so it renders as code. Ex: "-c" -> "`-c`"

File: _commands/basics/uniq.md [hunk 7]
```diff
@@ -2,23 +2,39 @@
 ---
 
 uniq
---
-TODO: Add documentation for this command by submitting a pull request.
-<!-- one line explanation would go here -->
+------
+
+`uniq`, often used with  `sort`, allows a user to report, or filter, repeated lines that are adjacent in a file. Some common and useful flags include -d or -u for listing lines that are duplicated or that are unique, respectfully. As well as -c for counting the number of times a line was dupicated.
```
Comment (cameron-gagnon) [inline _commands/basics/uniq.md]: The line from "Some common..." onward can be moved below the "`<!--more-->`" comment. That way the amount of content that appears on the reference page is minimal.

File: _commands/basics/uniq.md [hunk 8]
```diff
@@ -2,23 +2,39 @@
...
+##### Break it down
 
 #### Example command
-
+~~~ bash
+$ sort in.txt | uniq -d
+line1
+line2
+~~~
 ##### Break it down
```
Comment (cameron-gagnon) [inline _commands/basics/uniq.md]: Remove this line if there are no more examples after it.

**Comment by umdalema** [comment]
I believe I've corrected all the formatting errors. I've also added documentation of the -i and -f flags

**Pull request merged**